tests/framework/extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -1,19 +1,44 @@
 package etcdsnapshot
 
 import (
+	"github.com/rancher/norman/types"
 	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
 	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
 	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
 	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/sirupsen/logrus"
 )
 
 const (
 	ProvisioningSteveResouceType = "provisioning.cattle.io.cluster"
 	fleetNamespace               = "fleet-default"
+	active                       = "active"
 )
 
+func MatchNodeToAnyEtcdRole(client *rancher.Client, clusterID string) (int, *management.Node) {
+	machines, err := client.Management.Node.List(&types.ListOpts{Filters: map[string]interface{}{
+		"clusterId": clusterID,
+	}})
+	if err != nil {
+		return 0, nil
+	}
+
+	numOfNodes := 0
+	lastMatchingNode := &management.Node{}
+
+	for _, machine := range machines.Data {
+		if machine.Etcd {
+			lastMatchingNode = &machine
+			numOfNodes++
+		}
+	}
+
+	return numOfNodes, lastMatchingNode
+}
+
 // CreateSnapshot is a helper function to create a snapshot on an RKE2 or k3s cluster. Returns error if any.
-func CreateSnapshot(client *rancher.Client, clusterName string) error {
+func CreateSnapshot(client *rancher.Client, clusterName, clusterID string) error {
```

Comment:
Why is this taking a clusterID, but then not being used?

Reply:
I'll remove, that is a remnant from past debugging I was doing while making this PR.